src/stream/src/task/stream_manager.rs

```diff
@@ -179,35 +183,37 @@ impl LocalStreamManager {
     /// Broadcast a barrier to all senders. Returns when the barrier is fully collected.
     pub async fn send_and_collect_barrier(
```

Comment:
The changes in this function look weird to me. IIUC, we want to achieve two things:
- Respond back to meta with InjectBarrierResponse when barrier has been inserted into the source barrier channel. In other words, send back InjectBarrierResponse when `self.send_barrier` returns.
- Collect barrier asynchronously and sends ReportBarrierRequest when collection is done.

That being said, I think we can organize the code so that barrier sending and barrier collection is also separated:
- Add an new function named `initiate_barrier_collection` that takes the `rx` from `LocalStreamManager::send_barrier` and spawns an async task to do the collection and sends back ReportBarrierRequest.
- In `StreamServiceImpl::inject_barrier`, call `LocalStreamManager::send_barrier` and `LocalStreamManager::initiate_barrier_collection` separately.

Reply:
Yes, I think so too.I will modify it

Reply:
It looks strange that this async function returns another `Future`... +1 for splitting it into two stages.